Java-Introduction.md

```diff
@@ -0,0 +1,123 @@
+
+## Installation
+
+Download the latest Java binaries from the [official website](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html). Here you may face a question, which one to download, JDK or JRE? JRE stands for Java Runtime Environment, which is the platform dependent Java Virtual Machine to run Java codes, and JDK stands for Java Development Kit, which consists of most of the development tools, most importantly the compiler `javac`, and also the JRE. So, for an average user JRE would be sufficient, but since we would be developing with Java, we would download the JDK.
+
+Platform specific installation instructions
+
+### Windows
+
+* Download the relevant .msi file (x86 / i586 for 32bits, x64 for 64bits)
+* Run the .msi file. Its a self extracting executable file which will install Java in your system!
+
+### Linux
+
+* Download the relevant tar.gz file for your system and install :
+
+   ```bash
+   $ tar zxvf jdk-8uversion-linux-x64.tar.gz
+   ```
+* RPM based Linux platforms download the relevant .rpm file and install :
+
+   ```bash
+   $ rpm -ivh jdk-8uversion-linux-x64.rpm
+   ```
+* Users have the choice to install an open source version of Java, OpenJDK or the Oracle JDK. While OpenJDK is in active development and in sync with Oracle JDK, they just differ in licensing stuff, however few developers complain of the stability of Open JDK. Instructions for **Ubuntu** :
+
+   Open JDK installation :
+   ```bash
+   sudo apt-get install openjdk-8-jdk
+   ```
+
+   Oracle JDK installation :
+   ```bash
+   sudo add-apt-repository ppa:webupd8team/java
+   sudo apt-get update
+   sudo apt-get install oracle-java8-installer
+   ```
+
+### Mac
+
+* Either download Mac OSX .dmg executable from Oracle Downloads
+* Or use [Homebrew](http://brew.sh/) to [install](http://stackoverflow.com/a/28635465/2861269) :
+
+   ```bash
+   brew tap caskroom/cask
+   brew install brew-cask
+   brew cask install java
+   ```
+
+### Verify Installation
+
+Verify Java has been properly installed in your system by opening Command Prompt / Terminal and checking the versions of Java runtime and compiler :
+
+```bash
+$ java -version
+java version "1.8.0_66"
+Java(TM) SE Runtime Environment (build 1.8.0_66-b17)
+Java HotSpot(TM) 64-Bit Server VM (build 25.66-b17, mixed mode)
+
+$ javac -version
+javac 1.8.0_66
+```
+
+## JVM
+
+Ok now since we are done with the installations, lets begin to understand first the nitty gritty of the Java ecosystem. Java is an interpreted language, that is the code we write gets compiled and then run. We write the code in .java files, Java compiles them into [bytecodes](https://en.wikipedia.org/wiki/Java_bytecode) which are run on a Java Virtual Machine or JVM for execution. These bytecodes typically has a .class extension.
```

Comment:
> Java is an interpreted language

I understand you explain how the JVM interprets the bytecode right after this. But most Java people would take offense with this.

I suggest having this statement and the following one linked to some well-researched article. If possible, let's have it reviewed by a Java veteran.

I say this, because one of the salient point of Java is the fact that it is a compiled language.